doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -550,12 +562,166 @@ container sonic-interface {
 ......
 ```
 
+#### Example 2: Composite Keys with Same Number of Elements(NOT Allowed case)
```

Comment:
This one looks right to me.

Reply:
you mean Rule holds good.. And example given makes sense. correct?

Reply:
Hi @praveen-li , here is the issue created by @faraazbrcm : https://github.com/sonic-net/sonic-buildimage/issues/16356
The same key count cannot be parsed in GCU side. because two list share the same key count and same key type. When GCU is verifying the YANG model: [telemetry_client.yang](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-telemetry_client.yang#L68). GCU failed to parse TELEMETRY_CLIENT_SUB_LIST. It verify the list matches TELEMETRY_CLIENT_DS_LIST then return failure.

I would suggest to use different key count to protect GCU parse.

Reply:
I can think of two fix on this: 
Combine two list into one with key count=2. Example: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang#L70-L71
Use two different key count to represent these lists. Example: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-interface.yang#L45-L95

Reply:
Composite Keys with Same Number of Elements. 

[PC]: Same number of element is allowed by translation. 'Same number of element plus Same type' is not allowed. But if we can support patterns, we can allow same type in future. I feel it should be allowed and should be added in translation.

Reply:
@praveen-li Elements with distinct types but the same number can also lead to overlaps, especially since Redis always treats keys as strings. For instance, the number 123 can also be interpreted as the string '123'. To prevent this, we should apply specific constraints, like patterns for strings and ranges for integer types. This requirement should be clearly stated in the guideline.

Reply:
@praveen-li , in the telemetry client sample, the list defination(`TELEMETRY_CLIENT_SUB_LIST` and `TELEMETRY_CLIENT_DS_LIST`) belongs to the "same number of element plus Same type". It should not be allowed.